src/async_impl/client.rs

```diff
@@ -93,6 +93,7 @@ struct Config {
     cookie_store: Option<cookie::CookieStore>,
     trust_dns: bool,
     error: Option<crate::Error>,
+    error_for_status: bool,
```

Comment:
Would an enum make sense here rather than a bool?
https://rust-lang.github.io/api-guidelines/type-safety.html#c-custom-type

The name seems a bit like it hides the intention a bit. `call_error_for_status_on_server_error` or something like that might make sense. (Bikeshed opinion, feel free to ignore)